solutions/ys-SetXORMin.mdx

```diff
@@ -0,0 +1,105 @@
+---
+id: ys-SetXORMin
+source: YS
+title: Set XOR-Min
+author: Chongtian Ma
+---
+
+## Explanation
+
+We can construct a binary trie, with nodes representing the most significant bits near the root down to the least significant bits. When we add or remove a number fo a trie, we can simply add or subtract nodes with its binary representation. To answer queries, we can go down the trie and pick the same bit if possible to minimize total xor. Note when we go down any path in the trie, it is guaranteed that it will lead to the bottom so we don't have to worry about backtracking.
+
+
+## Implentation
+
+**Time Complexity:** $\mathcal{O}(q\log x)$
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <set>
+using namespace std;
+
+struct trie {
+	trie* children[2] = {};
+	int cnt = 0;
+};
+
+void add(trie* root, int x) {
```

Comment:
ever heard of property methods?

Reply:
erm doesnt rly matter

Reply:
it's good practice

Reply:
ok but like it really doesnt matter in this simple case

Reply:
i dont think itll work out in this case lol

Reply:
why not

Reply:
nastier implem

Reply:
@Aadit-Ambadkar until this is addressed i won't merge

Reply:
@SansPapyrus683 im not really sure what you mean. I think this version is fine but if you want to edit it with your suggestion feel free.

Reply:
i mean do you know have a method belong to a class?

Reply:
yeah, but im traversing on the object itself. I think i'd have to make the function recursive for that to work which might make it harder to read :/